### PR TITLE
Update llm README.md

### PR DESCRIPTION
**Pull request by jason-dai**
Update the README for `bigdl-llm`

### CODE REVIEW

**Comment by jason-dai** [comment]
@shane-huang @TheaperDeng please take a look

File: python/llm/README.md [hunk 1]
```diff
@@ -1,140 +1,137 @@
...
-#### langchain integration
-TODO
-
-## Examples
-We prepared several examples in https://github.com/intel-analytics/BigDL/tree/main/python/llm/example
-
-## Dynamic library BOM
-To avoid difficaulties during the installtion. `bigdl-llm` release the C implementation by dynamic library or executive file. The compilation details are stated below. **These information is only for reference, no compilation procedure is needed for our users.** `GLIBC` version may affect the compatibility.
+### `bigdl-llm` Dependence 
+The native code/lib in `bigdl-llm` has been built using the following tools; in particular, lower  `LIBC` version on your Linux system may be incompatible with `bigdl-llm`.
```
Comment (jason-dai) [inline python/llm/README.md:135]: What is the info for starcoder? @rnwang04

File: python/llm/README.md [hunk 2]
```diff
@@ -1,140 +1,137 @@
...
+
+#### Run Model
+ 
+You may run the models using **`bigdl-llm`** through one of the following APIs:
+1. [CLI (command line interface) Tool](#cli-tools)
+2. [Hugging Face `transformer`-style API](#hugging-face-transformers-style-api)
+3. LangChain API
+4. `llama-cpp-python`-style API
+
+#### CLI Tools
```
Comment (jason-dai) [inline python/llm/README.md:29]: please verify the convert and cli command @hoshibara @leonardozcm

File: python/llm/README.md (hunk 1 above)
Comment (rnwang04) [reply python/llm/README.md:135]: info is listed here: https://github.com/intel-analytics/BigDL/pull/8379/files#diff-83b9b2e754ffba10a70a073b16878e1a9f368822ffdb1e4cffc3df70dee84831
Have not merged this PR as action error.

File: python/llm/README.md [hunk 3]
```diff
@@ -1,140 +1,137 @@
...
+  #run the optimized model
+  from transformers import LlamaTokenizer
+  tokenizer = LlamaTokenizer.from_pretrained(model_path)
+  input_ids = tokenizer.encode(input_str, ...)
+  output_ids = model.generate(input_ids, ...)
+  output = tokenizer.decode(output_ids, ...)
+  ```
+
+  See the example here.
+
```
Comment (jason-dai) [inline python/llm/README.md:103]: please take a look @yangw1234

File: python/llm/README.md [hunk 4]
```diff
@@ -1,140 +1,137 @@
...
+You may download any PyTorch model in Hugging Face *Transformers* format (including *FP16* or *FP32* and *GPTQ-4bit*).
+
+#### Run Model
+ 
+You may run the models using **`bigdl-llm`** through one of the following APIs:
+1. [CLI (command line interface) Tool](#cli-tools)
+2. [Hugging Face `transformer`-style API](#hugging-face-transformers-style-api)
+3. LangChain API
+4. `llama-cpp-python`-style API
```
Comment (TheaperDeng) [inline python/llm/README.md:27]: should we add jump link for 3 and 4?

File: python/llm/README.md [hunk 5]
```diff
@@ -144,3 +141,5 @@ To avoid difficaulties during the installtion. `bigdl-llm` release the C impleme
 | gptneox      | Windows  | MSVC 19.36.32532.0 |       |
 | bloom        | Linux    | GCC 9.4.0          | 2.31  |
 | bloom        | Windows  | MSVC 19.36.32532.0 |       |
+| starcoder    | Linux    | GCC 9.4.0          | 2.31  |
+| starcoder    | Windows  | MSVC 19.36.32532.0 |       |
```
Comment (TheaperDeng) [inline python/llm/README.md:145]: some building parameters has been forked. We may need to rearrange this by split "convert", "inference" requirement for each family @Oscilloscope98

File: python/llm/README.md (hunk 2 above)
Comment (leonardozcm) [reply python/llm/README.md:29]: > please verify the convert and cli command @hoshibara @leonardozcm

llm-convert works fine after https://github.com/intel-analytics/BigDL/pull/8395

File: python/llm/README.md (hunk 5 above)
Comment (TheaperDeng) [reply python/llm/README.md:145]: please change 2.31->2.29 for both bloom and starcoder. We will use https://github.com/intel-analytics/BigDL/pull/8399 for a more detailed table later.

Comment (Oscilloscope98) [reply python/llm/README.md:145]: For gptneox and llama, Linux GCC should be updated to 9.3.1

File: python/llm/README.md [hunk 6]
```diff
@@ -1,140 +1,137 @@
...
+   #llama/bloom/gptneox/starcoder model family is currently supported
+   llm-cli -t 16 -x llama -m "/path/to/output/model.bin" -p 'Once upon a time,'
+   ```
+   
+#### Hugging Face `transformers`-style API
+You may run the models using `transformers`-style API in `bigdl-llm`
+
+- ##### Using native INT4 format
+
+   You may convert Hugging Face *Transformers* models into native INT4 format for maximum performance as follows. (Currently only *llama*/*bloom*/*gptneox*/*starcoder* model family is supported; for other models, you may use the [Hugging Face `transformers` format](hugging-face-transformers-style-api)).
```
Comment (TheaperDeng) [inline python/llm/README.md:65]: for other models, you may use the (This link should be pointed to #using-hugging-face-transformers-format ?)

File: python/llm/README.md [hunk 7]
```diff
@@ -1,140 +1,137 @@
...
+
+  You may apply INT4 optimizations to any Hugging Face *Transformers* models as follows.
+
+  ```python
+  #load Hugging Face Transformers model with INT4 optimizations
+  from bigdl.llm.transformers import AutoModelForCausalLM
+  model = AutoModelForCausalLM.from_pretrained('/path/to/model/', load_in_4bit=True)
+
+  #run the optimized model
+  from transformers import LlamaTokenizer
```
Comment (yangw1234) [inline python/llm/README.md:95]: Maybe 
```python
from transformers import AutoTokenizer
```
is more appropriate.

(The reason why `LlamaTokenizer` is specifically used in the example is that the tokenizer class name in "decapoda-research/llama-7b-hf" is wrong.)

File: python/llm/README.md [hunk 8]
```diff
@@ -1,140 +1,137 @@
...
+  tokenizer = LlamaTokenizer.from_pretrained(model_path)
```
Comment (yangw1234) [inline python/llm/README.md:96]: also `AutoTokenizer`

File: python/llm/README.md [hunk 9]
```diff
@@ -1,140 +1,137 @@
...
+  input_ids = tokenizer.encode(input_str, ...)
+  output_ids = model.generate(input_ids, ...)
+  output = tokenizer.decode(output_ids, ...)
```
Comment (yangw1234) [inline python/llm/README.md:99]: Maybe `tokenizer.decode(output_ids[0])` or `tokenizer.batch_decode(output_ids)`

File: python/llm/README.md (hunk 7 above)
Comment (TheaperDeng) [reply python/llm/README.md:95]: actually there is another reason. `AutoTokenizer` is SUPER slow for llama tokenizer (~10min loading) while LlamaTokenizer is much better

**Comment by shane-huang** [comment]
LGTM

**Pull request closed without merging**